src/index.js

```diff
@@ -3,247 +3,277 @@
   Author Tobias Koppers @sokra
 */
 
-// TODO: temporarily disabled rules
-/* eslint-disable
-  no-undefined,
-  no-param-reassign,
-  no-underscore-dangle,
-  import/order
-*/
 import { SourceMapConsumer } from 'source-map';
-import { ConcatSource, RawSource, SourceMapSource } from 'webpack-sources';
-
+import { SourceMapSource, RawSource, ConcatSource } from 'webpack-sources';
 import RequestShortener from 'webpack/lib/RequestShortener';
 import ModuleFilenameHelpers from 'webpack/lib/ModuleFilenameHelpers';
+import uglify from 'uglify-es';
 
-import uglify from 'uglify-js';
+/* eslint-disable
+  no-param-reassign
+*/
+
+const warningRegex = /\[.+:([0-9]+),([0-9]+)\]/;
+
+const defaultUglifyOptions = {
+  output: {
+    comments: /^\**!|@preserve|@license|@cc_on/,
+    beautify: false,
+    semicolons: true,
+    shebang: true,
+  },
+};
 
 class UglifyJsPlugin {
   constructor(options) {
     if (typeof options !== 'object' || Array.isArray(options)) {
-      options = {};
+      this.options = {};
+    } else {
+      this.options = options || {};
     }
-    if (typeof options.compressor !== 'undefined') {
-      options.compress = options.compressor;
+
+    this.options.test = this.options.test || /\.js($|\?)/i;
+    this.options.warningsFilter = this.options.warningsFilter || (() => true);
+
+    this.uglifyOptions = this.options.uglifyOptions || {};
```

Comment:
1.) The plugin should consume configurations, not offer them. Too many edge cases and more maintenance then you would think. The behavior of the plugin configuration should remain consistent with the existing major version range in this regard.

2.) Use `uglifyOptions `.